experimental/bits/simd_fixed_size.h

```diff
@@ -1331,6 +1340,56 @@ template <int _Np>
 	});
       }
 
+    // _S_gather {{{2
+    template <typename _Tp, typename _Up>
+      static inline _SimdMember<_Tp>
+      _S_gather(const _Up* __mem, const __int_for_sizeof_t<_Up>* __idx,
+		_TypeTag<_Tp>) noexcept
+      {
+	return _SimdMember<_Tp>::_S_generate([&](auto __meta) {
+	  return __meta._S_gather(__mem, &__idx[__meta._S_offset],
+				  _TypeTag<_Tp>());
+	});
+      }
+
+    // _S_gather {{{2
+    template <typename _Tp, typename _Up>
+      static inline _SimdMember<_Tp>
+      _S_gather(const _Up* __mem, const _SimdMember<_Tp>& __idx,
+		_TypeTag<_Tp>) noexcept
+      {
+	return _SimdMember<_Tp>::_S_generate([&](auto __meta) {
+	  return __meta._S_gather(
+	    __mem, __idx.template _M_tuple_at<__meta._S_offset>(),
+	    _TypeTag<_Tp>());
+	});
+      }
```

Comment:
Did you intend the types like this? I.e. the `value_type` of the index vector equals the `value_type` of the returned / gathered vector? I.e. you can't gather `_SimdMember<float>` unless `__idx` is of type `_SimdMember<float>`. I'd expect `int` for the latter.
Once you allow different types for gathered vector and index vector you will get size mismatches: e.g. gather floats using 64-bit int index vector. Or `double` / `int`, or `float` / `int` on AVX w/o AVX2. Then the `_M_tuple_at` access will fail. I hate this case, but I already handled it for some of the math functions. You should be able to find inspiration there. :wink: